crypto/param_build.c

```diff
@@ -305,6 +307,8 @@ OSSL_PARAM *ossl_param_bld_to_param(OSSL_PARAM_BLD *bld, void **secure)
     }
     if (secure != NULL)
         *secure = s;
+    if (sec_sz != NULL)
+        *sec_sz = ss;
```

Comment:
If I may be picky, the assignment of `*secure` and `*sec_sz` could be moved inside the `if (ss > 0)` block, and for those where we require non-NULL anyway, you could remove the ckeck.

Reply:
Double free anyone?
They also need to be assigned when secure memory isn't being allocated.

Reply:
> Double free anyone?

Huh?

Reply:
>  They also need to be assigned when secure memory isn't being allocated.

Does it?  Why?

Reply:
> Does it? Why?

D'oh, ignore that!  I need more coffee

Reply:
Yes, a lot more coffee.  I've got a tree you can harvest next time you are over....